### PR TITLE
Homepage&notes

### PR DESCRIPTION
**Pull request by suryxks**
Implemented:

-  Home page which displays the notes added 
- add notes
- edit notes
- pin and unpin notes

### CODE REVIEW

**Review by shraddha15-creator — APPROVED** [review]
UI looks good.
Code is clean and readable.

File: src/contexts/DataContext.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        } catch (e) {
+            console.error(e);
+        }
+    }
+    const editNote=async(encodedToken,note,noteId)=>{
+        try{
+         const {data}=await axios.post(`/api/notes/${noteId}`,{note},{ headers: { authorization: encodedToken } });
+         setNotes(data.notes)
+        }catch(e){
+
```
Comment (manojasarsa) [inline src/contexts/DataContext.jsx]: I think you have not handled the `catch` case here

File: src/pages/Home.jsx [hunk 2]
```diff
@@ -1,6 +1,92 @@
...
+  })
+  return (<PageContainer isNotesModalOpen={isNotesModalOpen}>
+    <NotesHeader >
+      <HeadingLarge>Notes</HeadingLarge>
+      <PageActions>
+        <FilterButton><FilterAltOutlinedIcon /><span>Filter</span></FilterButton>
+        <ButtonCta onClick={() => { setIsNotesModalOpen(true) }}><AddIcon /> <span>Add Note</span></ButtonCta>
+      </PageActions>
+    </NotesHeader>
+    {/* <NotesdModal isModalOpen={isNotesModalOpen} setIsModalOpen={setIsNotesModalOpen} note={currentNote} setNote={setCurrentNote} /> */}
```
Comment (manojasarsa) [inline src/pages/Home.jsx]: You can remove these comments if no longer needed

**Review by manojasarsa — APPROVED** [review]
Rest, code looks clean. GTG!

File: src/contexts/DataContext.jsx (hunk 1 above)
Comment (suryxks) [reply src/contexts/DataContext.jsx]: thanks for pointing it manoj fixed it

File: src/pages/Home.jsx (hunk 2 above)
Comment (suryxks) [reply src/pages/Home.jsx]: removed it , thanks

**Pull request merged**